cp1616/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(cp1616)
+
+find_package(catkin REQUIRED COMPONENTS 
+  roscpp
+  message_generation
+  )
+
+include_directories(${catkin_INCLUDE_DIRS})
+include_directories(${PROJECT_SOURCE_DIR}/include/)
+
+
+SET(SOURCES ${PROJECT_SOURCE_DIR}/src/cp1616_io_controller.cpp
+            ${PROJECT_SOURCE_DIR}/src/cp1616_io_controller_callbacks.cpp
+            ${PROJECT_SOURCE_DIR}/src/cp1616_io_device.cpp
+            ${PROJECT_SOURCE_DIR}/src/cp1616_io_device_callbacks.cpp
+            ${PROJECT_SOURCE_DIR}/src/cp1616_main.cpp
+          )
+
+SET(HEADERS ${PROJECT_SOURCE_DIR}/include/cp1616/cp1616_io_controller.h
+            ${PROJECT_SOURCE_DIR}/include/cp1616/cp1616_io_controller_callbacks.h
+            ${PROJECT_SOURCE_DIR}/include/cp1616/cp1616_io_device.h
+            ${PROJECT_SOURCE_DIR}/include/cp1616/cp1616_io_device_callbacks.h
+	  )
+	  
+catkin_package(
+  DEPENDS
+  INCLUDE_DIRS
+  CATKIN_DEPENDS roscpp 
+  LIBRARIES
+  roscpp
+)
+
+add_executable(${PROJECT_NAME} ${SOURCES} ${HEADERS})
```

Comment:
This package should create and executable and a library.  Only the ROS node should be compiled into the executable.  For now you can log an issue for this (no need to fix it now).